unit_tests/ngp_algorithms/UnitTestNodalGradPOpenBoundary.C

```diff
@@ -0,0 +1,122 @@
+// Copyright 2019 National Technology & Engineering Solutions of Sandia, LLC
+// (NTESS), National Renewable Energy Laboratory, University of Texas Austin,
+// Northwest Research Associates. Under the terms of Contract DE-NA0003525
+// with NTESS, the U.S. Government retains certain rights in this software.
+//
+// This software is released under the BSD 3-clause license. See LICENSE file
+// for more details.
+//
+
+#include "kernels/UnitTestKernelUtils.h"
+
+#include "UnitTestHelperObjects.h"
+
+#include "ngp_algorithms/UnitTestNgpAlgUtils.h"
+#include "ngp_algorithms/NodalGradPOpenBoundaryAlg.h"
+#include "ngp_algorithms/GeometryBoundaryAlg.h"
+#include "ngp_algorithms/NodalGradAlgDriver.h"
+#include "ngp_algorithms/GeometryAlgDriver.h"
+
+TEST_F(LowMachKernelHex8Mesh, NGP_nodal_grad_popen)
+{
+  // Only execute for 1 processor runs
+  if (bulk_.parallel_size() > 1) return;
+
+  const std::string meshSpec = "generated:20x20x20";
```

Comment:
What is the motivation to do this for a `20x20` face instead of a reference element?

Reply:
In order to have the ngp exactly match the non-ngp and give better coverage.  It may be overkill but the 20x20 runs in less than a second.

Reply:
I wasn't worried about the runtime. I guess I am unclear why a 1x1 face (or a 3x3) won't match the non-ngp exactly. Also, I was curious if you had found some differences in results that didn't show up in a 1x1 or a 3x3 face that we should look into with other algorithms also.

Reply:
I'm curious about this too. I agree that running 8000 elements is probably not a significant runtime issue, but is it the case that 1x1 or 2x2 gave non-matching results? That would seem to be unlikely...

Reply:
The question arose as to why a 20x20 test mesh and not 1x1 or 2x2.  I wanted a 3x3 or larger  because in this case the number of exposed faces makes a difference in the answer so I just wanted a variety of element exposed faces to get varied answers at sides, edges and corners.  As Shreyas pointed out below there is still a major bug that is not caught by this unit test.  In general I worried that a single element unit test might not be enough.

Reply:
@overfelt Because of the dual volume approach, the number of element sub-control faces connected to a node will accumulate different values to an interior node vs a boundary node along an edge or at a corner. However, the contributions to a node per-element should still be  the same.